src/runtime/policy/policy.ts

```diff
@@ -0,0 +1,270 @@
+      private readonly annotationRefs: AnnotationRef[]) {
+    this.validateAllowedUsages();
+    // TODO(b/157605585): Validate field structure against Type.
+  }
+
+  get allowedUsages() {
+    // TODO(b/157605585): Handle "raw" label separately?
+    return this.annotationRefs
+        .filter(annotation => annotation.name === 'allowedUsage')
+        .map(annotation => ({
+          usage: toAllowedUsageEnum(annotation.params['usageType']),
+          label: annotation.params['label'] as string,
+        }));
+  }
+
+  get annotations() {
+    return this.annotationRefs.filter(annotation => annotation.name !== 'allowedUsage');
+  }
+
+  private validateAllowedUsages() {
+    const usages: Map<string, Set<PolicyAllowedUsageType>> = new Map();
+    this.annotationRefs.filter(annotation => annotation.name === 'allowedUsage').forEach(annotation => {
+      const label = annotation.params['label'] as string;
+      const usageTypeParam = annotation.params['usageType'] as string;
+      const usageType = toAllowedUsageEnum(usageTypeParam);
+      if (!usages.has(label)) {
+        usages.set(label, new Set());
+      }
+      const usageTypes = usages.get(label);
+      if (usageTypes.has(usageType)) {
+        throw new Error(`Usage of label '${label}' for usage type '${usageTypeParam}' has already been allowed.`);
+      }
+      usageTypes.add(usageType);
+    });
+  }
+
+  toManifestString(): string {
+    const lines = this.annotationRefs.map(annotation => annotation.toString());
+    if (this.subfields.length) {
+      lines.push(
+        `${this.name} {`,
+        ...flatMap(this.subfields, field => indentLines(field.toManifestString())),
+        '}');
+    } else {
+      lines.push(`${this.name},`);
+    }
+    return lines.join('\n');
+  }
+
+  static fromAstNode(
+      node: AstNode.PolicyField,
+      buildAnnotationRefs: (ref: AstNode.AnnotationRef[]) => AnnotationRef[]): PolicyField {
+    const annotationRefs = buildAnnotationRefs(node.annotationRefs);
+
+    // Convert subfields.
+    checkNamesAreUnique(node.subfields);
+    const subfields = node.subfields.map(field => PolicyField.fromAstNode(field, buildAnnotationRefs));
+
+    return new PolicyField(node.name, subfields, annotationRefs);
+  }
+}
+
+class PolicyConfig {
+  constructor(readonly name: string, readonly metadata: Map<string, string>) {}
+
+  toManifestString(): string {
+    const lines = [`config ${this.name} {`];
+    for (const [k, v] of this.metadata) {
+      lines.push(`  ${k}: '${v}'`);
+    }
+    lines.push('}');
+    return lines.join('\n');
+  }
+
+  static fromAstNode(node: AstNode.PolicyConfig): PolicyConfig {
+    return new PolicyConfig(node.name, node.metadata);
+  }
+}
+
+/** Converts the given string to a value of PolicyAllowedUsageType. */
+// tslint:disable-next-line: no-any
+function toAllowedUsageEnum(value: any): PolicyAllowedUsageType {
+  switch (value) {
+    case '*':
+      return PolicyAllowedUsageType.Any;
```

Comment:
optional: in the number you could declare string values for each element and make the conversion more generic.

Reply:
Done.